sdk/ai/azure-ai-projects/README.md

```diff
@@ -327,6 +330,95 @@ agent = project_client.agents.create_agent(
+
+We can upload file to Azure as it is shown in the example, or use the existing Blob. In the code below we demonstrate how this can be achieved. First we are upload file to azure and create `VectorStoreDataSource`, which then is used to create vector store. This vector sore is then given to the `FileSearchTool` constructor.
+
+<!-- SNIPPET:sample_agents_enterprise_file_search.upload_file_and_create_agent_with_file_search -->
+
+```python
+# We will upload the local file to Azure and will use it for vector store creation.
+_, asset_uri = project_client.upload_file("./product_info_1.md")
+
+# create a vector store with no file and wait for it to be processed
+ds = VectorStoreDataSource(asset_identifier=asset_uri, asset_type=VectorStoreDataSourceAssetType.URI_ASSET)
+vector_store = project_client.agents.create_vector_store_and_poll(data_sources=[ds], name="sample_vector_store")
+print(f"Created vector store, vector store ID: {vector_store.id}")
+
+# create a file search tool
+file_search_tool = FileSearchTool(vector_store_ids=[vector_store.id])
+
+# notices that FileSearchTool as tool and tool_resources must be added or the assistant unable to search the file
+agent = project_client.agents.create_agent(
+    model="gpt-4-1106-preview",
+    name="my-assistant",
+    instructions="You are helpful assistant",
+    tools=file_search_tool.definitions,
+    tool_resources=file_search_tool.resources,
+)
+```
+
+<!-- END SNIPPET -->
+
+We also can attach files to the existing vector vector store. In the code snippet below, we first create empty vector store and add file to it.
+
+<!-- SNIPPET:sample_agents_vector_store_batch_enterprise_file_search.attach_files_to_store -->
+
+```python
+# create a vector store with no file and wait for it to be processed
+vector_store = project_client.agents.create_vector_store_and_poll(data_sources=[], name="sample_vector_store")
+print(f"Created vector store, vector store ID: {vector_store.id}")
+
+ds = VectorStoreDataSource(asset_identifier=asset_uri, asset_type=VectorStoreDataSourceAssetType.URI_ASSET)
+# add the file to the vector store or you can supply data sources in the vector store creation
+vector_store_file_batch = project_client.agents.create_vector_store_file_batch_and_poll(
+    vector_store_id=vector_store.id, data_sources=[ds]
+)
+print(f"Created vector store file batch, vector store file batch ID: {vector_store_file_batch.id}")
+
+# create a file search tool
+file_search_tool = FileSearchTool(vector_store_ids=[vector_store.id])
+```
+
+<!-- END SNIPPET -->
+
+We also can use Enterprise File Search without creation of a vector store. In this scenario we initialize the agent with `CodeInterpreterTool` definitions as shown below:
+
+<!-- SNIPPET:sample_agents_vector_store_batch_enterprise_file_search.create_agent -->
+
+```python
+code_interpreter = CodeInterpreterTool()
+
+# notice that CodeInterpreter must be enabled in the agent creation, otherwise the agent will not be able to see the file attachment
```

Comment:
code interpreter should not be required for file search

Reply:
According to the current README: 

> To attach a file to a message for data analysis, you use `MessageAttachment` and `CodeInterpreterTool`. You must pass `CodeInterpreterTool` as `tools` or `toolset` in `create_agent` call or the file attachment cannot be opened for code interpreter.  

Without code interpreter, the attachment cannot be read, here is what I am getting without it:
"I'm sorry, but I'm unable to directly access the content of files or attachments. However, if you can copy the text from the attachment and paste it here, I'll be able to help you with whatever you need."

With code interreter the response is more meaningful:
"The beginning of the file suggests that it's a text file that contains information about a product with `item_number: 1`. The text seems to be formatted, possibly using Markdown or a similar plain text formatting system, with headings and possibly other metadata.\n\nHere are the first few lines as read from the file..."

Reply:
Let's check this more, I will call

Reply:
Moved this section to the Code Interpreter as we have discussed offline.